### PR TITLE
feat(CAD): Add CAD to signin 

### LINKED ISSUE #5262: Expand Connect Another Device to all sync Desktop sign-ins
**Currently, in several situations, Sync users aren't seeing our great connect another device experience when they've verified a new account or confirm a sign-in.**

Here is the feature doc that @ryanfeeley , @shane-tomlinson and @davismtl worked on for this:
https://docs.google.com/document/d/1I2lspQj1x7AMIahM7PgcphnMSfLX-pJC_teru4HYaBs/edit#heading=h.q9y5y6jxb0qy

Here is the feature card for this work:
https://projects-beta.growthhackers.com/cards/add-connect-another-device-to-signin/ghpyETxuFuK63zGPZsC4kw

## Current
   

With a small addition of copy, we can roll out our Connect Another Device campaign to all, this campaign is safe to roll out to all remaining Sync authentications.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #5262

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 9ebc64955c572c6feb60d7775dad3dad21ab51b4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

File: app/scripts/views/mixins/connect-another-device-mixin.js [hunk 1]
```diff
@@ -47,18 +46,54 @@ define((require, exports, module) => {
...
+     */
+    isEligibleForConnectAnotherDeviceOnSignin (account) {
+      const isEligibleForCadOnSignin = !! this.getExperimentGroup('cadOnSignin', { account });
+
+      return this.isSignIn() &&
+             this.isEligibleForConnectAnotherDevice(account) &&
+             isEligibleForCadOnSignin;
+    },
+
+    navigateToConnectAnotherDeviceOnSigninScreen (account) {
```
Comment (shane-tomlinson) [inline app/scripts/views/mixins/connect-another-device-mixin.js]: Document this function.

File: tests/functional/fx_fennec_v1_force_auth.js [hunk 2]
```diff
@@ -28,22 +28,21 @@ define([
   const PASSWORD = 'password';
   let email;
 
-  const setupTest = thenify(function (options) {
```
Comment (shane-tomlinson) [inline tests/functional/fx_fennec_v1_force_auth.js]: Revert changes to this file, the experiment is not enabled for Fennec.

File: tests/functional/fx_fennec_v1_sign_in.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ define([
     return this.parent
       .then(clearBrowserState())
       .then(createUser(email, PASSWORD, { preVerified: options.preVerified }))
-      .then(openPage(SIGNIN_PAGE_URL, selectors.SIGNIN.HEADER))
+      .then(openPage(SIGNIN_PAGE_URL, selectors.SIGNIN.HEADER, { query: options.query }))
```
Comment (shane-tomlinson) [inline tests/functional/fx_fennec_v1_sign_in.js]: Revert changes to this file, the experiment is not enabled for Fennec.

File: tests/functional/fx_ios_v1_sign_in.js [hunk 4]
```diff
@@ -79,9 +79,9 @@ define([
         .then(clearBrowserState({ force: true }));
     },
 
-    'verified, verify same browser': function () {
```
Comment (shane-tomlinson) [inline tests/functional/fx_ios_v1_sign_in.js]: Revert changes to this file, the experiment is not enabled for Fx for iOS.

**Comment by shane-tomlinson** [comment]
This should be reviewed after #5447, but is ready for review!

@mozilla/fxa-devs - r?

**Comment by vladikoff** [comment]
Can be rebased :)

File: app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+      if (isEmailForcedIntoTreatment(subject.account.get('email'))) {
+        return 'treatment';
+      }
+
+      return false;
+    }
+
+    /**
+     * Are the subject pre-requisites met?
```
Comment (vladikoff) [inline app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js]: nit: `prerequisites` ?

File: app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+     *
+     * @param {Object} subject
+     * @returns {Boolean}
+     * @private
+     */
+    _areSubjectPrereqsMet (subject) {
```
Comment (vladikoff) [inline app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js]: nit: `_isValidSubject` ? :)

File: app/scripts/templates/connect_another_device.mustache [hunk 7]
```diff
@@ -4,13 +4,18 @@
     <div class="success success-authenticated visible">{{#t}}This Firefox is connected{{/t}}</div>
   {{/isSignedIn}}
   {{^isSignedIn}}
-    <div class="success success-not-authenticated visible">{{#t}}Email verified{{/t}}</div>
+    {{#isSignUp}}
+      <div class="success success-not-authenticated visible">{{#t}}Email verified{{/t}}</div>
+    {{/isSignUp}}
+    {{#isSignIn}}
+      <div class="success success-not-authenticated visible">{{#t}}Signin confirmed{{/t}}</div>
```
Comment (vladikoff) [inline app/scripts/templates/connect_another_device.mustache]: neat!

File: app/scripts/templates/connect_another_device.mustache [hunk 8]
```diff
@@ -4,13 +4,18 @@
...
+    {{/isSignIn}}
   {{/isSignedIn}}
   <section>
```
Comment (vladikoff) [inline app/scripts/templates/connect_another_device.mustache]: nit: maybe you want to just add a class to this section, something like `cad-instructions` and then css it with `.cad instructions p`.  Nit though!

File: app/scripts/templates/connect_another_device.mustache [hunk 9]
```diff
@@ -64,7 +78,7 @@
       {{/isFirefoxIos}}
     {{/canSignIn}}
     <div class="links">
-      <a data-flow-event="link.why" href="/connect_another_device/why">{{#t}}Why is this required?{{/t}}</a>
+      <a data-flow-event="link.why" href="/connect_another_device/why">{{#t}}Why is more than one device required?{{/t}}</a>
```
Comment (vladikoff) [inline app/scripts/templates/connect_another_device.mustache]: 👍

File: app/scripts/templates/sms_send.mustache [hunk 10]
```diff
@@ -6,7 +6,12 @@
     <section>
       <div class="graphic graphic-connect-another-device">{{#t}}Success{{/t}}</div>
 
-      <p>{{#t}}Send Firefox directly to your smartphone and sign in to complete set-up{{/t}}</p>
+      <p class="instructions">
```
Comment (vladikoff) [inline app/scripts/templates/sms_send.mustache]: but I guess we need it here too. maybe rethink the name `instructions`

File: app/scripts/views/base.js [hunk 11]
```diff
@@ -823,6 +824,15 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Has the view already navigated?
```
Comment (vladikoff) [inline app/scripts/views/base.js]: could you add a small doc about what this is useful / used for?

File: app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js (hunk 6 above)
Comment (shane-tomlinson) [reply app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js]: Much better than the name I used.

File: app/tests/spec/lib/experiments/grouping-rules/q3-form-changes.js [hunk 12]
```diff
@@ -24,15 +24,6 @@ define(function (require, exports, module) {
         assert.isFalse(experiment.choose({}));
       });
 
-      it('returns experiment if forceExperiment', () => {
```
Comment (shane-tomlinson) [inline app/tests/spec/lib/experiments/grouping-rules/q3-form-changes.js]: This should have been removed in master.

File: app/scripts/templates/connect_another_device.mustache (hunk 8 above)
Comment (shane-tomlinson) [reply app/scripts/templates/connect_another_device.mustache]: >  maybe you want to just add a class to this section, something like cad-instructions and then css it with .cad instructions p. Nit though!

You know... I originally added the class to add some styling. Turns out I didn't add any styling. Removing the extra class, avoiding the naming problem all together!

**Comment by shane-tomlinson** [comment]
@vladikoff - updated!

File: app/scripts/lib/experiments/grouping-rules/q3-form-changes.js [hunk 13]
```diff
@@ -20,7 +20,7 @@ define((require, exports, module) => {
      * @returns {Any}
      */
     choose (subject) {
-      const EXPERIMENTS = ['disabledButtonState', 'signupPasswordConfirm', 'emailFirst'];
+      const EXPERIMENTS = ['disabledButtonState', 'signupPasswordConfirm', 'emailFirst', 'cadOnSignin'];
```
Comment (vladikoff) [inline app/scripts/lib/experiments/grouping-rules/q3-form-changes.js]: 👍

File: app/scripts/views/mixins/connect-another-device-mixin.js [hunk 14]
```diff
@@ -47,18 +41,60 @@ define((require, exports, module) => {
     ],
 
     /**
+     * Is `account` eligible for he connect another device on signin?
```
Comment (vladikoff) [inline app/scripts/views/mixins/connect-another-device-mixin.js]: typo `he connect `

File: app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (vladikoff) [inline app/scripts/lib/experiments/grouping-rules/connect-another-device-on-signin.js]: Looks good, left one typo nit:

image

**Pull request merged**